### DISCUSSION TITLE
Can I manually add pages to gatsby-plugin-sitemap?

### DISCUSSION THREAD

**Discussion by jodiedoubleday** [gatsbyjs/gatsby#37697] [Help]
I have some software that loads/creates pages via JS that sit on my site. I want to add those pages to my sitemap even though they aren't generated by gatsby.

Can anybody tell me a way of manually adding those pages to the gatsby-plugin-sitemap? we have exclude, but what I need is include.

**Comment by gh640** [comment]
It'll help to read the package README:

- https://www.gatsbyjs.com/plugins/gatsby-plugin-sitemap/

I guess you need to use `resolvePages` in the options.

> resolvePages ⇒ Array
> 
> This allows custom resolution of the array of pages. This also where users could merge multiple sources into a single array if needed. Sync or async functions allowed.

If it doesn't work well as you expect, I recommend to read the code of the package. It's short and straightforward.

- https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-sitemap/src/gatsby-node.js

**Reply by jodiedoubleday** [reply]
Thanks for your reply. I have indeed looked at the README.

resolvePages - Takes the output of the data query and expects an array of page objects to be returned. Sync or async functions allowed.
Which can be seen in the example - https://www.gatsbyjs.com/plugins/gatsby-plugin-sitemap/#example

The pages I want to add aren't in any data query they are produced separately outside of gatsby.  So I want to manually add some URLs.

**Reply by gh640** [reply]
Thanks for your response. I don't fully understood "manually" means but pages we want to add don't need to be in data query. `resolvePages()` can return urls that are not in the arguments (=query) if I understand it correctly. Or, if you want, you can create dummy nodes that store urls that were produced outside of Gatsby so that `resolvePages()` can use them.

(Sorry if I've misunderstood something.)
